src/compiler/types.ts

```diff
@@ -5249,7 +5254,7 @@ namespace ts {
         getCompilerOptions(): CompilerOptions;
 
         /** Starts a new lexical environment. */
-        startLexicalEnvironment(): void;
+        startLexicalEnvironment(scoping?: LexicalEnvironmentScoping): void;

```

Comment:
I'm not sure this should be part of the public API. It presents the possibility of a custom transformer introducing a `let`-scoped variable we won't be able to properly down-level since it wasn't visited by the type checker.

Reply:
The new code changes don't expose a public API for starting a block scope and rely on the checker flags being set to determine which scope it should hoist a variable into.